src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -33,7 +33,7 @@ internal static class SyntaxNodeRemover
 
             var residualTrivia = remover.ResidualTrivia;
 
-            if (residualTrivia.Count > 0)
+            if (result != null && residualTrivia.Count > 0)
```

Comment:
when do you get a null result?

Reply:
When removing the [root element](https://github.com/dotnet/roslyn/pull/23165/commits/dc5568b3f3a3dfb4a67ced53f410d29c8dd5ae69#diff-0e0e5cee589dd5835bc357a2c0406c46R2554).

Reply:
it's useful in cases like that to add a comment to help the next person who comes along to understand :)

Reply:
I agree, added :)

Reply:
you didn't change Visit method, so, this is fixing existing but not related to your work, right?

Reply:
Correct, this check along with the correction in [AsRootOfNewTreeWithOptionsFrom](https://github.com/dotnet/roslyn/pull/23165/files#diff-a306823eb511d5213112898cc4681e6bR432) method fixes a NRE when removing the root node and is not directly related to this PR (can be splitted to another PR if desired).

Reply:
I am fine but @dotnet/roslyn-compiler ultimately will decide.